python/tank/util/user_settings.py

```diff
@@ -36,101 +42,168 @@ def _init_singleton(self):
         """
         Singleton initialization.
         """
-        path = self._compute_config_location()
-        logger.debug("Reading user settings from %s" % path)
+        self._path = self._compute_config_location()
+        logger.debug("Reading user settings from %s" % self._path)
 
-        self._user_config = self._load_config(path)
+        self._user_config = self._load_config(self._path)
 
         # Log the default settings
-        logger.debug("Default site: %s" % (self.default_site or "<missing>",))
-        logger.debug("Default login: %s" % (self.default_login or "<missing>",))
+        logger.debug("Default site: %s", self._to_friendly_str(self.default_site))
+        logger.debug("Default login: %s", self._to_friendly_str(self.default_login))
```

Comment:
Logging text this way is safer. If you forget a value when using % an error will be thrown, while using the logger and `,` an error will be logged instead with the origin of the erroneous call. Since we sometimes log in parts of the code that can't be easily (or are not) unit tested, it's safer to log this way, so it won't break the code if an error occurred.

Also, technically, if you use `,` instead of `%`, the formatting will be done only if the log level is actually enabled, so debug strings won't get formatted most of the time, saving some cycles.